query-engine/connectors/query-connector/src/interface.rs

```diff
@@ -262,6 +262,19 @@ pub trait WriteOperations {
         trace_id: Option<String>,
     ) -> crate::Result<usize>;
 
+    /// Inserts many records at once into the database and returns their
+    /// selected fields.
+    /// This method should not be used if the connector does not support
+    /// returning created rows.
+    async fn create_records_returning(
+        &mut self,
+        model: &Model,
+        args: Vec<WriteArgs>,
+        skip_duplicates: bool,
+        selected_fields: FieldSelection,
+        trace_id: Option<String>,
+    ) -> crate::Result<ManyRecords>;
+
```

Comment:
We discussed the options, this approach is widening the interface, the alternative. Which is passing options to `create_records` to support the two variants, and change the result type to be an enum, is not idea either.

I don't have a strong preference, the first case favors segregation of responsibilities, easier to understand as the new function has a simpler code path down the line. It requires additional implementation for each implementation (SQL and MongoDB Connection, and ConnectorTransaction structs).  

This is an easy refactor down the line, so I won't bother with details go with this one, and change the implementation of the decision was not what you wanted.